src/Core/Domain/Address/Command/EditManufacturerAddressCommand.php

```diff
@@ -0,0 +1,328 @@
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Address\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Address\Exception\AddressConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Address\ValueObject\AddressId;
+
+/**
+ * Edits manufacturer address
+ */
+class EditManufacturerAddressCommand
+{
+    /**
+     * @var AddressId
+     */
+    private $addressId;
+
+    /**
+     * @var int
+     */
+    private $manufacturerId;
+
+    /**
+     * @var string|null
+     */
+    private $lastName;
+
+    /**
+     * @var string|null
+     */
+    private $firstName;
+
+    /**
+     * @var string|null
+     */
+    private $address;
+
+    /**
+     * @var string|null
+     */
+    private $city;
+
+    /**
+     * @var string|null
+     */
+    private $address2;
+
+    /**
+     * @var int|null
+     */
+    private $countryId;
+
+    /**
+     * @var string|null
+     */
+    private $postCode;
+
+    /**
+     * @var int|null
+     */
+    private $stateId;
+
+    /**
+     * @var string|null
+     */
+    private $homePhone;
+
+    /**
+     * @var string|null
+     */
+    private $mobilePhone;
+
+    /**
+     * @var string|null
+     */
+    private $other;
+
+    /**
+     * @param int $addressId
+     *
+     * @throws AddressConstraintException
+     */
+    public function __construct($addressId)
+    {
+        $this->addressId = new AddressId($addressId);
+    }
+
+    /**
+     * @return AddressId
+     */
+    public function getAddressId()
+    {
+        return $this->addressId;
+    }
+
+    /**
+     * @param AddressId $addressId
+     */
+    public function setAddressId($addressId)
+    {
+        $this->addressId = $addressId;
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getManufacturerId()
+    {
+        return $this->manufacturerId;
+    }
+
+    /**
+     * @param int $manufacturerId
+     *
+     * @throws AddressConstraintException
+     */
+    public function setManufacturerId($manufacturerId)
+    {
+        if (!is_int($manufacturerId) || 0 > $manufacturerId) {
```

Comment:
Should not we use again a `assertIsNullOrNonNegativeInt()` method ? An Edit Address Command cannot reset the manufacturer ID ?